client.go

```diff
@@ -53,6 +66,45 @@ func Timeout(timeout time.Duration) func(*Client) error {
 	}
 }
 
+// Keepalive keeps the connection open.
+func Keepalive() func(*Client) error {
+	return CustomKeepalive(DefaultKeepAlive)
+}
+
+// CustomKeepalive is the same as Keepalive() with a custom interval.
+func CustomKeepalive(interval time.Duration) func(*Client) error {
+	return func(c *Client) error {
+		go func(c *Client) {
+			for c.IsConnected() {
+				time.Sleep(interval)
+
+				c.mutex.Lock()
+				if err := c.setDeadline(); err != nil {
+					break
+				}
+				if _, err := c.conn.Write([]byte(" \n")); err != nil {
+					break
+				}
+				if err := c.clearDeadline(); err != nil {
+					break
+				}
+				c.mutex.Unlock()
+			}
+
+			c.setDisconnected()
+		}(c)
+		return nil
+	}
+}
+
+// NotificationBuffer sets the notification buffer size.
+func NotificationBuffer(size int) func(*Client) error {
+	return func(c *Client) error {
+		c.notify = make(chan Notification, size)
```

Comment:
Still not as clean is could be, if you have a bufferSize on the client with a default value then once you have run the options you just unconditionally create the channel.